File: MAP/lab7/lab7/Utils/ExeStack.cs
using System.Collections;
using System.Collections.Generic;
using System.Linq;

namespace lab7.Utils
{
    public class ExeStack<T> : IExeStack<T>
    {
        private Stack<T> stack = new Stack<T>();


        public void push(T x)
        {
            stack.Push(x);
        }

        public T pop()
        {
            return stack.Pop();
        }

        public bool isEmpty()
        {
            return stack.Count == 0;
        }

        public IEnumerator GetEnumerator()
        {
            return stack.GetEnumerator();
        }
    }
}

File: MAP/lab8 - cdLatch/src/utils/ExeStack.java
package utils;

import java.util.ArrayDeque;
import java.util.Deque;

public class ExeStack<T> implements IExeStack<T> {

    private Deque<T> stack = new ArrayDeque<>();

    @Override
    public void push(T x) {
         stack.push(x);
    }

    @Override
    public T pop() {

        return stack.pop();
    }

    @Override
    public boolean isEmpty() {
        return stack.size() == 0;
    }

    @Override
    public Iterable<T> getAll() {
        return stack;
    }

    @Override
    public String toString(){
        StringBuffer str = new StringBuffer();
        for(T i : stack){
            str.append(i);
        }
        return str.toString();
    }
}
